### ISSUE TITLE
Replace file() with open() in twisted/trial

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8520]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8520|
|Type|enhancement|
|Created|2016-06-28 06:17:30Z|
|Branch|https://github.com/twisted/twisted/tree/8520-rodrigc-file|

In Python 3, file() is gone (see https://docs.python.org/3/whatsnew/3.0.html#builtins ), and needs to be replaced with open()
which works on Python 2 and 3

<details><summary>Searchable metadata</summary>

```
trac-id__8520 8520
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__8520_rodrigc_file 8520-rodrigc-file
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1467094650865582 1467094650865582
changetime__1467101564160550 1467101564160550
version__None None
owner__rodrigc rodrigc

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_1"></a>|@rodrigc commented|
|-|-|

See:
https://github.com/twisted/twisted/pull/244

**Comment by twisted-trac** [comment]
|[<img alt="Lukasa's avatar" src="https://avatars.githubusercontent.com/u/1382556?s=50" width="50" height="50">](https://github.com/Lukasa)<a name="note_2"></a>|@Lukasa set owner to @rodrigc|
|-|-|

LGTM. The builders are misbehaving, but those failures are about the buildbots themselves, not this change.


Merge when ready.

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|GitHub set status to `closed`|
|-|-|

In changeset a241030f54307a976e4484cb56886430a4d6e664
```
#!CommitTicketReference repository="" revision="a241030f54307a976e4484cb56886430a4d6e664"
Merge 8520-rodrigc-file: Replace file() with open() in twisted/trial

Author: rodrigc
Reviewer: Lukasa
Fixes: #8520
```
